engine/api/router.go

```diff
@@ -219,50 +220,53 @@ func (r *Router) Handle(uri string, handlers ...RouterConfigParam) {
 		}
 
 		permissionOk := true
-		if rc.auth && rc.needAdmin && !c.User.Admin {
+		if rc.auth && rc.needHatchery && c.Hatchery == nil {
```

Comment:
c.Hatchery is set in func (r *Router) checkHatcheryAuth(db *gorp.DbMap, headers http.Header, c *context.Ctx) error 

needHatchery is set with func NeedHatchery() RouterConfigParam

NeedHatchery() is added on main_route